src/Credentials/CredentialProvider.php

```diff
@@ -267,77 +280,73 @@ public static function ecsCredentials(array $config = [])
+     * credentials from ini profile
+     *
+     * @param string|null $assumeRoleProfile   Assume role profile name to be used.
+     *                                         Default:
+     *                                         [assumes_role] in credentials file
+     *
+     * @param string|null $fileName            A custom file to fetch credential from.
+     *                                         Default: credential file in home directory
+     * @return callable
+     */
+    public static function assumeRole($assumeRoleProfile = null, $fileName = null)
+    {
+        $fileName = $fileName ?: (self::getHomeDir() . '/.aws/credentials');
+        $profile = $assumeRoleProfile ?: 'assumes_role';
+
+        return new AssumeRoleCredentialResolver([
+            'assume_role_profile' => $profile,
+            'file_name' => $fileName
+        ]);
+    }
+
+    /**
      * Credentials provider that creates credentials using an ini file stored
      * in the current user's home directory.
      *
-     * @param string|null $profile  Profile to use. If not specified will use
-     *                              the "default" profile.
-     * @param string|null $filename If provided, uses a custom filename rather
-     *                              than looking in the home directory for the
-     *
+     * @param string|null $profile      Profile to use. If not specified will use
+     *                                  the "default" profile.
+     * @param string|null $fileName     If provided, uses a custom filename rather
+     *                                  than looking in the home directory.
+     *                                  Default:
+     *                                  - 'default' in credential file
      * @return callable
      */
-    public static function ini($profile = null, $filename = null)
+    public static function ini($profile = null, $fileName = null)
     {
-        $filename = $filename ?: (self::getHomeDir() . '/.aws/credentials');
+        $fileName = $fileName ?: (self::getHomeDir() . '/.aws/credentials');
         $profile = $profile ?: (getenv(self::ENV_PROFILE) ?: 'default');
 
-        return function () use ($profile, $filename) {
-            if (!is_readable($filename)) {
-                return self::reject("Cannot read credentials from $filename");
-            }
-            $data = parse_ini_file($filename, true);
-            if ($data === false) {
-                return self::reject("Invalid credentials file: $filename");
-            }
-            if (!isset($data[$profile])) {
-                return self::reject("'$profile' not found in credentials file");
+        return function () use ($profile, $fileName) {
+            $msg = self::checkProfile($profile, $fileName);
+            if (!empty($msg)) {
+                return self::reject($msg);
             }
-            if (!isset($data[$profile]['aws_access_key_id'])
```

Comment:
Actually, the behavior stays the same. Because `checkProfile` is called first before `getProfileData`, it would have thrown exception already for this situation. Current test covers (this](https://github.com/aws/aws-sdk-php/blob/master/tests/Credentials/CredentialProviderTest.php#L235) already. Let me know if I misunderstood what you are suggesting :)